napari/_qt/widgets/qt_welcome.py

```diff
@@ -0,0 +1,181 @@
+
+    def __init__(self, parent):
+        super().__init__(parent)
+
+        # Create colored icon using theme
+        self._image = QLabel()
+        self._image.setObjectName("logo_silhouette")
+        self._image.setMinimumSize(300, 300)
+        self._label = QtWelcomeLabel(
+            trans._(
+                "Drag image(s) here to open\nor\nUse the menu shortcuts below:"
+            )
+        )
+
+        # Widget setup
+        self.setAutoFillBackground(True)
+        self.setAcceptDrops(True)
+        self._image.setAlignment(Qt.AlignCenter)
+        self._label.setAlignment(Qt.AlignCenter)
+
+        # Layout
+        text_layout = QVBoxLayout()
+        text_layout.addWidget(self._label)
+
+        # TODO: Use action manager for shortcut query and handling
+        shortcut_layout = QFormLayout()
+        sc = QKeySequence('Ctrl+O').toString(QKeySequence.NativeText)
+        shortcut_layout.addRow(
+            QtShortcutLabel(sc),
+            QtShortcutLabel(trans._("open image(s)")),
+        )
+        sc = QKeySequence('Ctrl+Alt+/').toString(QKeySequence.NativeText)
+        shortcut_layout.addRow(
+            QtShortcutLabel(sc),
+            QtShortcutLabel(trans._("show all key bindings")),
+        )
+        shortcut_layout.setSpacing(0)
+
+        layout = QVBoxLayout()
+        layout.addStretch()
+        layout.setSpacing(30)
+        layout.addWidget(self._image)
+        layout.addLayout(text_layout)
+        layout.addLayout(shortcut_layout)
+        layout.addStretch()
+
+        self.setLayout(layout)
+
+    def paintEvent(self, event):
+        """Override Qt method.
+
+        Parameters
+        ----------
+        event : qtpy.QtCore.QEvent
+            Event from the Qt context.
+        """
+        option = QStyleOption()
+        option.initFrom(self)
+        p = QPainter(self)
+        self.style().drawPrimitive(QStyle.PE_Widget, option, p, self)
+
+    def _update_property(self, prop, value):
+        """Update properties of widget to update style.
+
+        Parameters
+        ----------
+        prop : str
+            Property name to update.
+        value : bool
+            Property value to update.
+        """
+        self.setProperty(prop, value)
+        self.style().unpolish(self)
+        self.style().polish(self)
+
+    def dragEnterEvent(self, event):
+        """Override Qt method.
+
+        Provide style updates on event.
+
+        Parameters
+        ----------
+        event : qtpy.QtCore.QEvent
+            Event from the Qt context.
+        """
+        self._update_property("drag", True)
+        if event.mimeData().hasUrls():
+            event.accept()
+        else:
+            event.ignore()
+
+    def dragLeaveEvent(self, event):
+        """Override Qt method.
+
+        Provide style updates on event.
+
+        Parameters
+        ----------
+        event : qtpy.QtCore.QEvent
+            Event from the Qt context.
+        """
```

Comment:
```suggestion
    def dragLeaveEvent(self, event: QEvent):
        """Override dragLeaveEvent to provide style updates."""
```

we don't really have a "style guide" to point to here... and I'm not sure how everyone else feels about this, but with this sort of thing (where it's not a critical user facing thing that will show up in our sphinxdocs), I feel like it becomes much easier to visually parse the file if we boil it down to a type hint and a docstring.  (totally ok with this as it already is though!)

Reply:
I am fine with either option, it is just easier to enforce a "single rule" even if it results in a bit of more verbose docstings?

Reply:
very true :)